javalib/src/main/scala/java/lang/Byte.scala

```diff
@@ -39,30 +52,48 @@ object Byte {
    * against the JDK .class files anyway.
    */
   def MIN_VALUE: scala.Byte = -128
+
   def MAX_VALUE: scala.Byte = 127
 
-  @inline def valueOf(byteValue: scala.Byte): Byte = new Byte(byteValue)
-  @inline def valueOf(s: String): Byte             = valueOf(parseByte(s))
+  @inline def compare(x: scala.Byte, y: scala.Byte): scala.Int =
+    x - y
 
-  @inline def valueOf(s: String, radix: Int): Byte =
-    valueOf(parseByte(s, radix))
+  @inline def decode(nm: _String): Byte = {
+    val i = Integer.decode(nm).intValue
+    val b = i.toByte
+    if (b == i)
+      valueOf(b)
+    else
+      throw new NumberFormatException()
+  }
 
-  @inline def parseByte(s: String): scala.Byte = parseByte(s, 10)
+  @inline def hashCode(b: scala.Byte): scala.Int =
+    b.toInt
 
-  def parseByte(s: String, radix: Int): scala.Byte = {
-    val r = Integer.parseInt(s, radix)
-    if (r < MIN_VALUE || r > MAX_VALUE)
+  @inline def parseByte(s: _String, radix: scala.Int): scala.Byte = {
+    val i = Integer.parseInt(s, radix)
+    if (i < MIN_VALUE || i > MAX_VALUE)
       throw new NumberFormatException(s"""For input string: "$s"""")
     else
-      r.toByte
+      i.toByte
   }
 
-  @inline def toString(b: scala.Byte): String =
-    String.valueOf(b)
+  @inline def parseByte(s: _String): scala.Byte =
+    parseByte(s, 10)
 
-  @inline def compare(x: scala.Byte, y: scala.Byte): scala.Int =
-    x - y
+  @inline def toString(b: scala.Byte): _String =
+    Integer.toString(b)
 
-  @inline def hashCode(b: scala.Byte): scala.Int =
-    b
+//  def toUnsignedInt(x: scala.Byte): scala.Int =
+
+//  def toUnsignedLong(x: scala.Byte): scala.Long =
```

Comment:
Those two should have `undefined` as the body of the method (they'll are going to be intrinsified.)